Plenigo 659 support woocommerce payment

SebaSOFT opened the pull request:
REady for testing on the demo WP site...

Commit:
- Code complete, for the WooCommerce stuff, lets check that it actually works

Commit:
Better check for the WooCommerce stuff to be enabled

Commit:
- More fixes for aesthetics and functionality

Commit:
- Now we are doing everything on the checkout page... everything....

Commit:
- Now everything happens in the checkout page
- Now the (fixed) iframe starts when pushing a button, not automatically
- Now administrators HAVE to select product type for their products, seriously
- On plenigo bought check (considering  server -side paywall disabling) the order is marked as completed

Commit:
- Now everything happens in the checkout page
- Now the (fixed) iframe starts when pushing a button, not automatically
- Now administrators HAVE to select product type for their products, seriously
- On plenigo bought check (considering  server -side paywall disabling) the order is marked as completed

Commit:
- now services running at the fancy api.plenigo.com subdomain

robertholfeld commented:
Wrong comment [SettingCategoryTagDB](https://github.com/plenigo/plenigo_wordpress_plugin/blob/PLENIGO-659-support-woocommerce-payment/plenigo_plugin/settings/SettingCategoryTagDB.php#L24)

robertholfeld commented:
Wrong misspelling [SettingCategoryTagDB](https://github.com/plenigo/plenigo_wordpress_plugin/blob/PLENIGO-659-support-woocommerce-payment/plenigo_plugin/settings/SettingCategoryTagDB.php#L39) mdae

robertholfeld commented:
Comment missing [SettingCategoryTagDB](https://github.com/plenigo/plenigo_wordpress_plugin/blob/PLENIGO-659-support-woocommerce-payment/plenigo_plugin/settings/SettingCategoryTagDB.php#L149)
Why size 35? [SettingCompanyId](https://github.com/plenigo/plenigo_wordpress_plugin/blob/PLENIGO-659-support-woocommerce-payment/plenigo_plugin/settings/SettingCompanyId.php#L75)
For what stands this 5? [SettingCompanyId](https://github.com/plenigo/plenigo_wordpress_plugin/blob/PLENIGO-659-support-woocommerce-payment/plenigo_plugin/settings/SettingCompanyId.php#L84)

robertholfeld commented:
Stylied? [SettingCurtainMode](https://github.com/plenigo/plenigo_wordpress_plugin/blob/PLENIGO-659-support-woocommerce-payment/plenigo_plugin/settings/SettingCurtainMode.php#L97)

SebaSOFT commented:
Why size 35? SettingCompanyId
- To have more than the default width on the text box, this doesnt affect the string max length

For what stands this 5? SettingCompanyId
- For a minimum ammount of character to be considered valid, I dont know any other rules that apply so I put some basic checks

Stylied? SettingCurtainMode
- Stylized

Commit:
- Robert pointed some misspelings, totally un related to the branch features, ROFL

robertholfeld commented:
- no-Javacript [SettingNoscriptMessage](https://github.com/plenigo/plenigo_wordpress_plugin/blob/PLENIGO-659-support-woocommerce-payment/plenigo_plugin/settings/SettingNoscriptMessage.php#L55)
- no-Javacript [SettingNoscriptTitle](https://github.com/plenigo/plenigo_wordpress_plugin/blob/PLENIGO-659-support-woocommerce-payment/plenigo_plugin/settings/SettingNoscriptTitle.php#L53)

Commit:
- Revert back to using order ID

Commit:
- goodbye my beloved Javacript, hello Javascript

robertholfeld commented on plenigo_plugin/settings/SettingCategoryTagDB.php:
```diff
@@ -36,7 +36,7 @@ class SettingCategoryTagDB extends PlenigoWPSetting
 
     /**
-     * Holds values for the SQL requests, so they are mdae just once per request
+     * Holds values for tstoring values, so they are generated just once per request
```
tstoring? you dont want to fix this?

SebaSOFT replied:
done

robertholfeld commented on plenigo_plugin/settings/SettingWooOrderTitle.php:
```diff
@@ -62,6 +62,7 @@ public function renderCallback() {
             . '<b>%%SITE_TITLE%%</b> - The Site&quot;s title.<br>'
             . '<b>%%ORDER_ID%%</b> - This is the order ID.<br>'
             . '<b>%%ORDER_NO%%</b> - This is the order number (maybe same as ID, depending on other plugins).<br>'
+            . '<b>%%ORDER_KEY%%</b> - This is the order identifier key (usuario <i>wc_order_XXXXXXXXXX</i>).<br>'
```
usuario?

robertholfeld commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -59,8 +61,10 @@ public function __construct() {
-        $this->description = __('WooCommerce', self::PLENIGO_SETTINGS_GROUP);
+        $this->title = __('Plenigo Payment', self::PLENIGO_SETTINGS_GROUP);
+        $this->method_title = __('Plenigo Payment', self::PLENIGO_SETTINGS_GROUP);
+        $this->description = __('WooCommerce Plenigo Description', self::PLENIGO_SETTINGS_GROUP);
```
plenigo and description lower case

SebaSOFT replied:
just when not starting a sentence

robertholfeld commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -59,8 +61,10 @@ public function __construct() {
+        $this->title = __('Plenigo Payment', self::PLENIGO_SETTINGS_GROUP);
+        $this->method_title = __('Plenigo Payment', self::PLENIGO_SETTINGS_GROUP);
+        $this->description = __('WooCommerce Plenigo Description', self::PLENIGO_SETTINGS_GROUP);
+        $this->method_description = __('WooCommerce Plenigo Description', self::PLENIGO_SETTINGS_GROUP);
```
plenigo and description lower case

SebaSOFT replied:
Just when not staring a sentence

robertholfeld commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -70,39 +74,26 @@ public function __construct() {
+        $order = new \WC_Order($order_id);
 
         // Mark as on-hold (we're awaiting the payment)
         $order->update_status('on-hold', __('Awaiting Plenigo payment', self::PLENIGO_SETTINGS_GROUP));
```
plenigo

SebaSOFT replied:
Done

robertholfeld commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -111,41 +102,138 @@ public function process_payment($order_id) {
+                    plenigo_log_message("WOO: Checkout snippet:" . var_export($checkoutSnippet, true), E_USER_NOTICE);
+                    echo '<div style="width:100%;text-align:right;">'
+                    . '<button class="checkout-button button alt wc-forward" onclick="' . $checkoutSnippet . '">'
+                    . __('Continue to Plenigo checkout', self::PLENIGO_SETTINGS_GROUP)
```
plenigo lower case

SebaSOFT replied:
done

robertholfeld commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -111,41 +102,138 @@ public function process_payment($order_id) {
+        if ($this->is_valid_for_use()) {
+            parent::admin_options();
+        } else {
+            ?>
```
?> is this correct? I See no answer!!!!

SebaSOFT replied:
It was ok but I did it differently to be more clear

robertholfeld commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -111,41 +102,138 @@ public function process_payment($order_id) {
+            parent::admin_options();
+        } else {
+            ?>
+            <div class="inline error"><p><strong><?php _e('Gateway Disabled', self::PLENIGO_SETTINGS_GROUP); ?></strong>: <?php _e('Please complete the configuration in the Plenigo settings.', self::PLENIGO_SETTINGS_GROUP); ?></p></div>
```
disabled and plenigo lower case

SebaSOFT replied:
Disabled is correct, other done

robertholfeld commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -111,41 +102,138 @@ public function process_payment($order_id) {
+        } else {
+            ?>
+            <div class="inline error"><p><strong><?php _e('Gateway Disabled', self::PLENIGO_SETTINGS_GROUP); ?></strong>: <?php _e('Please complete the configuration in the Plenigo settings.', self::PLENIGO_SETTINGS_GROUP); ?></p></div>
+            <?php
```
?

SebaSOFT replied:
?

robertholfeld commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -111,41 +102,138 @@ public function process_payment($order_id) {
+    }
+
+    /**
+     * Detectes if the icheckout process finished and then marks the order as paid if necessary
```
Really icheckout?

SebaSOFT replied:
done

robertholfeld commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -111,41 +102,138 @@ public function process_payment($order_id) {
+
+    /**
+     * Detectes if the icheckout process finished and then marks the order as paid if necessary
+     * @param int $order_id The order ID to check for pleigo user bought
```
pleigo?

SebaSOFT replied:
done

robertholfeld commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -111,41 +102,138 @@ public function process_payment($order_id) {
+            $order = new \WC_Order($order_id);
+            $user_bought = \plenigo_plugin\PlenigoSDKManager::get()->plenigo_bought($order->id);
+            if ($user_bought === true) {
+                plenigo_log_message("WOO: Uer bouight it with Plenigo!", E_USER_NOTICE);
```
bouight? plenigo lower case

robertholfeld commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -111,41 +102,138 @@ public function process_payment($order_id) {
+                $order->payment_complete();
+            } else {
+                // Here could be maybe a payment timeout to set it as cancelled
+                plenigo_log_message("WOO: Uer DID NOT buy this with Plenigo...yet?!", E_USER_NOTICE);
```
Uer? plenigo lower case

robertholfeld commented on readme.txt:
```diff
@@ -129,6 +129,9 @@ separator, native Wordpress tag.
 
 == Changelog ==
+= 1.1.32 - WooCommerce Payment Gateway =
+- Feature: Now you can use Plenigo as a WooCommerce Payment Gateway! No further setup required! Altough you can customize the order title to be shown in the curtomers bills. Start selling your products right away!
```
plenigo lower case, curtomers?

SebaSOFT replied:
done

Commit:
- Using the new "partial redirect URL" feature

Commit:
usuario...who typed that?

robertholfeld commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -34,6 +33,8 @@ class WC_Gateway_Plenigo extends \WC_Payment_Gateway {
      */
     private $options = null;
+    // Available Product Types
+    private $prodTypeList = array('EBOOK', 'DIGITALNEWSPAPER', 'DOWNLOAD', 'VIDEO', 'MUSIC');
```
Thats the same as in SettingWooProductType

SebaSOFT replied:
Yes but resuding it will require and import and requiring the File, I can consider a single static class for constants and fixed values like this.....currently distributed in all classes

robertholfeld replied:
I understand that a import of a third party file is not good, but when it is a file of our project it should not be a problem, or i am wrong?

SebaSOFT replied:
The thing is to iport a classwith unrelated functionality just because we havea const/field that we could reuse...

robertholfeld commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -111,41 +102,139 @@ public function process_payment($order_id) {
+                $order->update_status('processing', __('Plenigo checkout stating', self::PLENIGO_SETTINGS_GROUP));
+                plenigo_log_message("WOO: Creating checkout snippet:", E_USER_NOTICE);
+
+                //Let's create a unmanaged Plenigo Product for this order
```
plenigo product lower case

Commit:
- Plenigo SDK updated
- Fixed problems with detectng payments fixed
- Now using the "Awaiting Payment" status
- Welcome WooCommerce!

Commit:
- spelling and typo correctiong

maximilianschweitzer commented on plenigo_plugin/PlenigoSDKManager.php:
```diff
@@ -106,6 +107,7 @@ public function get_csrf_token() {
      */
     public function plenigo_bought($products = null) {
         if (is_null($products)) {
+            plenigo_log_message("Plenigo bought check: false => products null", E_USER_WARNING);
```
Is warning really the right level here. This could happend all the time.

maximilianschweitzer commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -111,41 +127,135 @@ public function process_payment($order_id) {
+                $order->update_status('completed', __('Plenigo payment complete. Thank you!', self::PLENIGO_SETTINGS_GROUP));
+            } else {
+                // Here could be maybe a payment timeout to set it as cancelled
+                plenigo_log_message("WOO: Uer DID NOT buy this with plenigo...yet?!", E_USER_NOTICE);
```
Uer should be User I think.

maximilianschweitzer commented on plenigo_plugin/wooCommerce/WC_Gateway_Plenigo.php:
```diff
@@ -111,41 +127,135 @@ public function process_payment($order_id) {
+            $order = new \WC_Order($order_id);
+            $user_bought = \plenigo_plugin\PlenigoSDKManager::get()->plenigo_bought($order_id);
+            if ($user_bought === true) {
+                plenigo_log_message("WOO: Uer bouight it with plenigo!", E_USER_NOTICE);
```
Uer should be User I think.

maximilianschweitzer replied:
Uer should be User I think.

Commit:
- Uer => User

Commit:
- NOTICE level for some warnings
